test/allow.js

```diff
@@ -0,0 +1,131 @@
+const assert = require('chai').assert
+const { getUserId, verifyDelegator } = require('../lib/handlers/allow')
+const rdf = require('rdflib')
+const ns = require('solid-namespace')(rdf)
+
+const alice = 'https://alice.com'  // principal
+const agentWebId = 'https://agent.example.com'  // secretary
+
+const verifiedSecretaryGraph = rdf.graph()
+verifiedSecretaryGraph.add(
+    rdf.namedNode(alice),
+    ns.acl('delegates'),
+    rdf.namedNode(agentWebId)
+  )
+const notVerifiedSecretaryGraph = rdf.graph()
+
+describe('allow() handler', () => {
+  describe('getUserId()', () => {
+    const emptyFetchDocument = () => {}
+    it('no webId present', done => {
+      let webId  // no secretary
+      let onBehalfOf = alice  // principal
```

Comment:
Additionally, I expect the case in which the request has no webId but an on-behalf-of header to be a rejected promise, not for `getUserId` to resolve to `undefined`.

As an aside I'd generalize that I'd almost never expect a promise to resolve to `undefined`. Null or falsy values should be explicit.